### ISSUE TITLE
Namespace overrides object declaration

### ISSUE THREAD

**Issue by ernestostifano** [microsoft/TypeScript#37773]
**TypeScript Version:**  3.9.0-dev.20200328

**Search Terms:** object declaration, namespace overrides

**Expected behavior:**
Both object and namespace declarations to be present.

**Actual behavior:**
Only namespace declaration is present.

When `myObj` is imported somewhere else, compiler complains (Can't use namespace as a value), but there is a value!

If export is removed, then `const propA` and `const propB` declarations appear. `const myObj` still doesn't.

**Related Issues:** N/A.

**Code**
```ts
/** @typedef {(string|RegExp|Array<(string|RegExp)>)} myObj.typeA */

/**
 * @typedef myObj.typeB
 * @property {myObj.typeA} someValueA
 * @property {string} someValueB
 */

/** @typedef {(myObj.typeB|Function)} myObj.typeC */

const propA = true;

const propB = 0;

const myObj = {
    propA,
    propB
};

export { myObj };
```

<details><summary><b>Output</b></summary>

```ts
export namespace myObj {
    export type typeA = string | RegExp | (string | RegExp)[];
    export type typeB = {
        someValueA: string | RegExp | (string | RegExp)[];
        someValueB: string;
    };
    export type typeC = Function | typeB;
}

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": false,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": true,
    "allowJs": true,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play/?ts=3.9.0-dev.20200328&useJavaScript=true#code/PQKhAIAEBcE8AcCmATRAzcBvAFAZ2gE4CWAdgOYA+ASomQKIAe8FAggQQIawA8ehplGvSYBKAHwiAvuAC2sAPIAjAFYA6OEhbgQwAFC7QIXdqgaU6WQpXqEiAELGIkeAQD2SAnCxylasy2lcVxlEADUOABsAV0QWRygXd0RPWCx8YnJA4LDImIdtPQMwU1tUDBwfazM7CgAxKJIAY2giVxIpS18bJABhAv1GtvxwRPgtAF5wQhiAbgGh6BG3eDtwSYAGOd1BkmHK5TWsY3AT0ZYAGmPT5YdJLcQmV08sTpVwO6A)

**Comment by sandersn** [comment]
This doesn't repro for me. Here's the program I tried:

```json
{
    "compilerOptions": {
        "allowJs": true,
        "checkJs": true,
        "noEmit": true
    },
    "files": ["foo.js", "bar.js"]
}
```

```js
/** @typedef {(string|RegExp|Array<(string|RegExp)>)} myObj.typeA */

/**
 * @typedef myObj.typeB
 * @property {myObj.typeA} someValueA
 * @property {string} someValueB
 */

/** @typedef {(myObj.typeB|Function)} myObj.typeC */

const propA = true;

const propB = 0;

const myObj = {
    propA,
    propB
};

export { myObj };
```

```js
import { myObj } from "./foo";
myObj.propA = false
/** @type {myObj.typeA} */
var x = 1 //'hi'
```

`myObj` worked both as a value (`myObj.propA`) and as a namespace (`myObj.typeA`). Perhaps this got fixed since you filed the bug?

**Comment by sandersn** [comment]
Now that I got my Typescript versions sorted out, I can see that this is broken in 3.9 and fixed in 4.0.
